pyfaulttolerance/timeout.py

```
@@ -10,29 +10,29 @@
-            # If effective_seconds is 0 due to float truncation (e.g., seconds=0.1),
-            # signal.alarm(0) would cancel any pending alarm, effectively not setting a new one.
-            # This specific subtask is focused on seconds == 0.
-            # The behavior for 0 < seconds < 1 for this sync decorator means it won't timeout
-            # before the function completes, unless the function itself is extremely short.
+            # For positive timeouts, signal.alarm requires integers.
+            # If seconds is > 0 and < 1, int(seconds) would be 0,
+            # which cancels the alarm. So, use 1 second as the minimum effective timeout.
+            if 0 < seconds < 1:
+                effective_seconds = 1
+            else:
+                effective_seconds = int(seconds) # For seconds >= 1 or if seconds is a whole number float
```

Ruff (0.11.9)

Use ternary operator `effective_seconds = 1 if 0 < seconds < 1 else int(seconds)` instead of `if`-`else`-block

(SIM108)